### DISCUSSION TITLE
Not working trailing stop ?

### DISCUSSION THREAD

**Discussion by luki009** [conor19w/Binance-Futures-Trading-Bot#22] [General]
Hi,
looking and playing currently with your bot (btw: great job, looks really great) and noticed  your note about trailing stop not working.
I looked closer on the trailing stop order logic and noticed possible issue.
You are using  **"stopPrice"** parameter with  type='TRAILING_STOP_MARKET' but stopPrice parameter is not used by Binance API for that type. You should use **"activationPrice"** parameter instead with desired price like take profit. In that case trailing stop loss will be activated with selected callback which is calculated based on current price.
You can push activationPrice  as empty parameter which will cause that activation price will be current market price on order submit.
**NOTE:** Please consider this as suggestion fix and look on it. I was not able to test it yet.
See: https://binance-docs.github.io/apidocs/futures/en/#new-order-trade

<html>
<body>

activationPrice | DECIMAL | NO | Used with TRAILING_STOP_MARKET orders, default as the latest price(supporting different workingType)
-- | -- | -- | --
stopPrice | DECIMAL | NO | Used with STOP/STOP_MARKET or TAKE_PROFIT/TAKE_PROFIT_MARKET orders.

</body>
</html>
                                 
https://github.com/conor19w/Binance-Futures-Trading-Bot/blob/3ff7f4ea86ac564a2b613c718677d34707355949/Bot.py#L249-L267

Aditional note: 
I suggest to use "reduceOnly" or "closePosition" position parameter with TAKE_PROFIT/TAKE_PROFIT_MARKET, STOP/STOP_MARKET and TRAILING_STOP_MARKET. Preferably "reduceOnly" with 100% of trade size to prevent closing existing position and opening new trade in opposite way in case of  trade size miscalculation due to rounding, etc.
<html>
<body>

reduceOnly | STRING | NO | "true" or "false". default "false". Cannot be sent in Hedge Mode; cannot be sent with closePosition=true
-- | -- | -- | --
closePosition | STRING | NO | true, false；Close-All，used with STOP_MARKET or TAKE_PROFIT_MARKET.

</body>
</html>

</body>
</html>

**Comment by conor19w** [comment]
Good spot I've added your suggestion into Bot.py, Closeposition is used where possible already. Don't think it can be used with Limit orders
